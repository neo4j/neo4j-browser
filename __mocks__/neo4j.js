/*
 * Copyright (c) 2002-2018 "Neo4j, Inc"
 * Network Engine for Objects in Lund AB [http://neotechnology.com]
 *
 * This file is part of Neo4j.
 *
 * Neo4j is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

function integerFn (val) {
  this.val = val
}
integerFn.prototype.toString = function () {
  return this.val.toString()
}

var out = {
  v1: {
    Int: integerFn,
    isInt: function (val) {
      return val instanceof integerFn
    },
    types: {
      Node: function Node (id, labels, properties) {
        this.identity = id
        this.labels = labels
        this.properties = properties
      },
      Relationship: function Relationship (id, start, end, type, properties) {
        this.identity = id
        this.start = start
        this.end = end
        this.type = type
        this.properties = properties
      },
      Path: function Path (start, end, segments) {
        this.start = start
        this.end = end
        this.segments = segments
        this.length = segments.length
      },
      PathSegment: function PathSegment (start, relationship, end) {
        this.start = start
        this.relationship = relationship
        this.end = end
      }
    },
    Integer: function Integer ({ low, high }) {
      this.low = low
      this.high = high
    }
  }
}

out.v1.types.Node.prototype.toString = function () {
  return 'node'
}
out.v1.types.Relationship.prototype.toString = function () {
  return 'rel'
}
out.v1.types.Path.prototype.toString = function () {
  return 'path'
}
out.v1.types.PathSegment.prototype.toString = function () {
  return 'pathsegment'
}
out.v1.Integer.prototype.toInt = function () {
  return this.low
}
out.v1.int = val => new out.v1.Integer(val)

module.exports = out
