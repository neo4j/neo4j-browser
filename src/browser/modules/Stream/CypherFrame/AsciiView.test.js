/*
 * Copyright (c) 2002-2020 "Neo4j,"
 * Neo4j Sweden AB [http://neo4j.com]
 *
 * This file is part of Neo4j.
 *
 * Neo4j is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

import React from 'react'
import { render } from '@testing-library/react'
import neo4j from 'neo4j-driver'

import {
  AsciiViewComponent as AsciiView,
  AsciiStatusbarComponent as AsciiStatusbar
} from './AsciiView'

describe('AsciiViews', () => {
  describe('AsciiView', () => {
    test('displays bodyMessage if no rows', () => {
      // Given
      const sps = jest.fn()
      const result = {
        records: [],
        summary: {
          resultAvailableAfter: neo4j.int(5),
          resultConsumedAfter: neo4j.int(5)
        }
      }

      // When
      const { container } = render(
        <AsciiView setParentState={sps} result={result} maxRows={5} />
      )

      // Then
      expect(container).toMatchSnapshot()
    })
    test('does not display bodyMessage if rows', () => {
      // Given
      const sps = jest.fn()
      const result = {
        records: [{ keys: ['x'], _fields: ['y'], get: () => 'y' }],
        summary: {
          resultAvailableAfter: neo4j.int(5),
          resultConsumedAfter: neo4j.int(5)
        }
      }

      // When
      const { container } = render(
        <AsciiView setParentState={sps} result={result} />
      )

      // Then
      expect(container).toMatchSnapshot()
    })
  })
  describe('AsciiStatusbar', () => {
    test('displays statusBarMessage if no rows', () => {
      // Given
      const sps = jest.fn()
      const result = {
        records: [],
        summary: {
          resultAvailableAfter: neo4j.int(5),
          resultConsumedAfter: neo4j.int(5)
        }
      }

      // When
      const { container } = render(
        <AsciiStatusbar setParentState={sps} result={result} />
      )

      // Then
      expect(container).toMatchSnapshot()
    })
    test('displays statusBarMessage if no rows', () => {
      // Given
      const sps = jest.fn()
      const result = {
        records: [{ keys: ['x'], _fields: ['y'], get: () => 'y' }],
        summary: {
          resultAvailableAfter: neo4j.int(5),
          resultConsumedAfter: neo4j.int(5)
        }
      }
      const serializedRows = [['x'], ['y']]

      // When
      const { container } = render(
        <AsciiStatusbar
          setParentState={sps}
          result={result}
          _asciiSerializedRows={serializedRows}
        />
      )

      // Then
      expect(container).toMatchSnapshot()
    })
  })
})
