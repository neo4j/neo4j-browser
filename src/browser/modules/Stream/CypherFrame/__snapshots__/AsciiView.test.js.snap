// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AsciiViews AsciiStatusbar displays statusBarMessage if no rows 1`] = `
<div>
  <div
    class="styled__StyledStatsBar-sc-1dtvgs1-23 gWjkIU"
  >
    <div
      class="Ellipsis-sc-1dn9bou-0 gzqRPC"
    >
      Completed after 10  ms.
    </div>
  </div>
</div>
`;

exports[`AsciiViews AsciiStatusbar displays statusBarMessage if no rows 2`] = `
<div>
  <div
    class="styled__StyledStatsBar-sc-1dtvgs1-23 gWjkIU"
  >
    <div
      class="styled__StyledRightPartial-sc-1dtvgs1-33 kITufN"
    >
      <div
        class="styled__StyledWidthSliderContainer-sc-1dtvgs1-35 hsuISq"
      >
        Max column width:
        <input
          class="styled__StyledWidthSlider-sc-1dtvgs1-36 cgzxFB"
          max="3"
          min="3"
          type="range"
          value="70"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`AsciiViews AsciiView displays bodyMessage if no rows 1`] = `
<div>
  <div
    class="styled__PaddedDiv-sc-1dtvgs1-1 kyHRpW"
  >
    <div
      class="styled__StyledBodyMessage-sc-1dtvgs1-27 hpwxul"
    >
      (no changes, no records)
    </div>
  </div>
</div>
`;

exports[`AsciiViews AsciiView does not display bodyMessage if rows 1`] = `
<div>
  <div
    class="styled__PaddedDiv-sc-1dtvgs1-1 kyHRpW"
  >
    <pre>
      ╒═══╕
│"x"│
╞═══╡
│"y"│
└───┘
    </pre>
  </div>
</div>
`;
